### PR TITLE
MPI-4: Implement MPI_Info_create_env()

### PR DESCRIPTION
**Pull request by shintaro-iwasaki**
## Pull Request Description

This patch implements `MPI_Info_create_env()`, which will be introduced by MPI-4 (#4915).

Note `MPI_Info_create_env()` can be called before `MPI_Init()` or after `MPI_Finalize()` thanks to #4978.

## Expected Impact

None.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [ ] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] You or your company has a signed contributor's agreement on file with Argonne
* [x] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by shintaro-iwasaki** [comment]
test:mpich/ch4/most

**Comment by shintaro-iwasaki** [comment]
test:mpich/ch3/most

**Comment by shintaro-iwasaki** [comment]
test:mpich/ch4/most
test:mpich/ch3/most

**Comment by shintaro-iwasaki** [comment]
@hzhou I'd be happy if you could review the PR when you have time.

**Review by hzhou — CHANGES_REQUESTED** [review]
Looks good. Minor issues.

File: src/mpi/info/infoutil.c
Comment (shintaro-iwasaki) [reply src/mpi/info/infoutil.c:71]: Thanks! `MPII_init_local_proc_attrs()`, which is the other caller of `MPIR_Info_setup_env()`, this does not take `argc` and `argc` at this point. To be honest, I don't know what parameters will be needed to set up this info object; for example, Fortran passes `NULL` for both so perhaps such arguments will not be necessary to initialize this info object. Since now the created info object is empty, I believe we can keep it as it is.

File: src/mpi/info/info_impl.c
Comment (shintaro-iwasaki) [reply src/mpi/info/info_impl.c:274]: Thanks. You're right. I will move this change to the commit for C.

Comment (shintaro-iwasaki) [reply src/mpi/info/info_impl.c:274]: Fixed. (It might not be reflected on this page because of a GitHub "incident" (https://www.githubstatus.com/incidents/rk5rhrklkdk3))

**Comment by shintaro-iwasaki** [comment]
test:mpich/ch4/most

**Comment by shintaro-iwasaki** [comment]
Thanks @hzhou! Once it passes all the tests, I will merge this PR (possibly after rebasing it).

test:mpich/ch3/most

**Comment by shintaro-iwasaki** [comment]
It passed all the tests. I will merge this PR. Thank you for reviewing this PR @hzhou!

**Pull request merged**
